test/task.jl

```diff
@@ -32,6 +32,13 @@
         @test e isa Ray.RayRemoteException
         @test e.captured.ex == ErrorException("AHHHHH")
     end
+
+    # object refs as arguments
+    obj_ref1 = Ray.put(1)
+    obj_ref2 = Ray.submit_task(identity, (obj_ref1,))
+    @test obj_ref2 != obj_ref1
+    @test Ray.get(obj_ref2) == obj_ref1
+    @test Ray.get(Ray.get(obj_ref2)) == 1
```

Comment:
(even if it is possible it shoudl probably be a separate test...)

Reply:
I'm working on adding support for this. The test I have in mind is: 
```julia
# use object ref arguments in task
obj_ref = Ray.put(1)
result = Ray.submit_task(Ray.get, (obj_ref,))
@test result isa ObjectRef
@test result != obj_ref
@test Ray.get(result) == 1
```
See: https://github.com/beacon-biosignals/Ray.jl/issues/77#issuecomment-1710314498

I definitely do want to keep including this test as it only requires serialized `ObjectRef`s to work. It also requires keeping track of inlined references

Reply:
I guess the other thing at issue here is whether teh tasks can/should be doing automatic de-referencing of literal objectref arguments like this.  IIRC python _does_ do that but I could be wrong

Reply:
You are correct that Python does indeed fetch any object refs passed in as arguments automatically. This seems like a mistake to me because:

1. You may not be using the object ref passed to the Ray task (possibly passing it along to a subtask)
2. Makes dispatch harder to reason about.

Reply:
Okay, let's file a follow-up to decide on this.